sklearn/decomposition/_pca.py

```diff
@@ -619,12 +619,12 @@ def _fit_truncated(self, X, n_components, svd_solver):
 
         # Get variance explained by singular values
         self.explained_variance_ = (S**2) / (n_samples - 1)
-        total_var = np.var(X, ddof=1, axis=0)
-        self.explained_variance_ratio_ = self.explained_variance_ / total_var.sum()
+        total_var = in_place_feature_var(X).sum()
```

Comment:
Turns out X is already centered at this point, see l.593. So we can avoid this extra computation. Based on that I don't think it's necessary to define a function for these 3 lines.

```suggestion
        np.square(X, out=X)
        np.sum(X, axis=0, out=X[0])
        feature_var = X[0] / (X.shape[0] - 1)
        total_var = feature_var.sum()
```